src/heading.rs

```diff
@@ -0,0 +1,20 @@
+use kuchiki::NodeDataRef;
+
+use crate::html;
+use crate::spec::Spec;
+
+pub fn process_headings(doc: &mut Spec) {
+    // TODO:
+    // 1. [all] or [doc-only]?
+    // 2. [settled] or not?
+    if let Some(ref document) = doc.document {
```

Comment:
nit: I'd early-return here, so:

```
let document = match doc.document {
    Some(ref doc) => doc,
    None => return,
};

let headings = match document.select(..) { ... };
```

Though probably you can just unwrap() the select, as the selector we pass is static and known-valid.